contracts/src/CrossChainApplications/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -0,0 +1,167 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@subnet-evm-contracts/interfaces/IWarpMessenger.sol";
+import "@subnet-evm-contracts/interfaces/IAllowList.sol";
+import "@subnet-evm-contracts/interfaces/INativeMinter.sol";
+import "./INativeTokenDestination.sol";
+import "../../Teleporter/ITeleporterMessenger.sol";
+import "../../Teleporter/ITeleporterReceiver.sol";
+import "../../Teleporter/SafeERC20TransferFrom.sol";
+
+// Precompiled Native Minter Contract Address
+address constant MINTER_ADDRESS = 0x0200000000000000000000000000000000000001;
+// Designated Blackhole Address
+address constant BLACKHOLE_ADDRESS = 0x0100000000000000000000000000000000000000;
+
+contract NativeTokenDestination is
+    ITeleporterReceiver,
+    INativeTokenDestination,
+    ReentrancyGuard
+{
+    INativeMinter private immutable _nativeMinter =
+        INativeMinter(MINTER_ADDRESS);
+
+    address public constant WARP_PRECOMPILE_ADDRESS =
```

Comment:
Lets either put all of the contracts above the contract definition, or within it here. (Currently two above, one within)